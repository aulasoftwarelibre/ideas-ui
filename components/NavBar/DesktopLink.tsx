import Link from 'next/link';
import { NextRouter, withRouter } from 'next/router';
import React from 'react';

interface Props {
  label: string;
  path: string;
  router: NextRouter;
}

const DesktopLink: React.FunctionComponent<Props> = ({
  label,
  path,
  router,
}) => {
  const isActive: boolean = router.pathname === path;

  if (isActive) {
    return (
      <span className="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-ideaOrange">
        {label}
      </span>
    );
  }

  return (
    <Link href={path}>
      <a className="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange">
        {label}
      </a>
    </Link>
  );
};

export default withRouter(DesktopLink);
