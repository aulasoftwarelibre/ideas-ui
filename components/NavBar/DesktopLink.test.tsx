import { screen } from '@testing-library/react';
import React from 'react';

import { renderMockRouter } from '../../utils/test-utils';
import { Props } from './DesktopLink';
import { Active, Default } from './DesktopLink.stories';

describe("DesktopLink", () => {
  it("should be rendered correctly", () => {
    renderMockRouter(<Default {...(Default.args as Props)} />, {
      router: { pathname: "/active" },
    });

    expect(screen.getByTestId("desktop-link"));
  });

  it("should be rendered correctly when current is enabled", () => {
    renderMockRouter(<Active {...(Active.args as Props)} />, {
      router: { pathname: "/active" },
    });

    expect(screen.getByTestId("active-desktop-link"));
  });
});
