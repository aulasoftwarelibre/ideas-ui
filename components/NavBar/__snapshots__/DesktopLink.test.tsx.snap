// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopLink should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
    href="/no-active"
  >
    Profile
  </a>
</DocumentFragment>
`;

exports[`DesktopLink should be rendered correctly when current is enabled 1`] = `
<DocumentFragment>
  <span
    class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-primary"
  >
    Profile
  </span>
</DocumentFragment>
`;
